nucliadb_node/src/settings/mod.rs

```diff
@@ -30,291 +30,191 @@
 pub struct Settings {
-    // Data storage and access
-    #[builder(default = "\"data\".into()", setter(custom))]
-    data_path: PathBuf,
-    #[builder(private, default = "PathBuf::from(\"data\").join(METADATA_FILE)")]
-    metadata_path: PathBuf,
-    #[builder(private, default = "PathBuf::from(\"data\").join(SHARDS_DIR)")]
-    shards_path: PathBuf,
-    #[builder(default = "true", setter(custom))]
-    lazy_loading: bool,
-    #[builder(default = "800")]
-    max_shards_per_node: usize,
-
-    // Index node self data
-    #[builder(default = "\"host_key\".into()", setter(name = "host_key_path"))]
-    node_id_path: PathBuf,
-    #[builder(default = "reliable_lookup_host(\"localhost\").ip()", setter(custom))]
-    public_ip: IpAddr,
-
-    // Errors
-    #[builder(default = "true", setter(custom))]
-    sentry_enabled: bool,
-    #[builder(default = "String::new()")]
-    sentry_url: String,
-    #[builder(default = "SENTRY_DEV", setter(custom))]
-    sentry_env: &'static str,
-
-    // Logs and traces
-    #[builder(default = "parse_log_levels(\"nucliadb_*=INFO\")")]
-    log_levels: Vec<(String, Level)>,
-    #[rustfmt::skip]
-    #[builder(
-        default = "parse_log_levels(\"nucliadb_node=INFO,nucliadb_core=INFO\")",
-        setter(skip)
-    )]
-    span_levels: Vec<(String, Level)>,
-
-    #[builder(default = "false", setter(custom))]
-    plain_logs: bool,
-
-    // Telemetry
-    #[builder(default = "false", setter(custom))]
-    jaeger_enabled: bool,
-    #[builder(default = "\"localhost\".into()")]
-    jaeger_agent_host: String,
-    #[builder(default = "6831")]
-    jaeger_agent_port: u16,
-
-    #[builder(default = "reliable_lookup_host(\"localhost:40102\")", setter(custom))]
-    reader_listen_address: SocketAddr,
-    #[builder(default = "reliable_lookup_host(\"localhost:40101\")", setter(custom))]
-    writer_listen_address: SocketAddr,
-
-    #[builder(default = "3030")]
-    metrics_port: u16,
-
-    // replications settings
-    #[builder(default = "\"http://localhost:10000\".into()")]
-    primary_address: String,
-    #[builder(default = "parse_node_role(\"primary\")")]
-    node_role: NodeRole,
-    #[builder(default = "3")]
-    replication_delay_seconds: u64,
-    #[builder(default = "3")]
-    replication_max_concurrency: u64,
-
-    // number of seconds since last replication for
-    // node to be considered healthy
-    #[builder(default = "30")]
-    replication_healthy_delay: u64,
+    inner: Arc<InnerSettings>,
+}
+
+impl From<InnerSettings> for Settings {
+    fn from(value: InnerSettings) -> Self {
+        Settings {
+            inner: Arc::new(value),
+        }
+    }
+}
+
+impl Clone for Settings {
+    fn clone(&self) -> Self {
+        Settings {
+            inner: Arc::clone(&self.inner),
+        }
+    }
 }
 
 impl Settings {
-    pub fn builder() -> SettingsBuilder {
-        SettingsBuilder::default()
+    pub fn builder() -> InnerSettingsBuilder {
```

Comment:
`InnerSettingsBuilder` is exposed and are public. If you're hiding the implementation details from `Settings`, why expose `InnerSettings`?